src/net/rpc.rs

```diff
@@ -303,20 +307,90 @@ impl RPCPoxInfoData {
             .to_owned()
             .expect_u128() as u64;
 
-        let total_required = total_liquid_supply_ustx
-            .checked_div(rejection_fraction)
-            .expect("FATAL: unable to compute total_liquid_supply_ustx/current_rejection_votes");
+        let total_required = (total_liquid_supply_ustx as u128)
+            .checked_div(100)
+            .expect("FATAL: unable to compute total_liquid_supply_ustx * 100")
+            .checked_mul(rejection_fraction as u128)
+            .expect("FATAL: unable to compute total_liquid_supply_ustx/current_rejection_votes")
+            as u64;
+
         let rejection_votes_left_required = total_required.saturating_sub(current_rejection_votes);
 
         let burnchain_tip = SortitionDB::get_canonical_burn_chain_tip(sortdb.conn())?;
 
-        let next_reward_cycle_in = reward_cycle_length
-            - ((burnchain_tip.block_height - first_burnchain_block_height) % reward_cycle_length);
+        let pox_consts = &burnchain.pox_constants;
+
+        if prepare_cycle_length != pox_consts.prepare_length as u64 {
+            error!(
+                "PoX Constants in config mismatched with PoX contract constants: {} != {}",
+                prepare_cycle_length, pox_consts.prepare_length
+            );
+            return Err(net_error::DBError(db_error::Corruption));
+        }
+
+        if reward_cycle_length != pox_consts.reward_cycle_length as u64 {
+            error!(
+                "PoX Constants in config mismatched with PoX contract constants: {} != {}",
+                reward_cycle_length, pox_consts.reward_cycle_length
+            );
+            return Err(net_error::DBError(db_error::Corruption));
+        }
+
+        let effective_height = burnchain_tip.block_height - first_burnchain_block_height;
```

Comment:
These subtractions can _never_ underflow, right?

Reply:
Right, if the canonical burn chain tip is less than `first_burnchain_block_height`, there is something very wrong happening.